src/main/java/com/aws/greengrass/componentmanager/ComponentServiceHelper.java

```diff
@@ -84,14 +85,21 @@ ResolvedComponentVersion resolveComponentVersion(String componentName, Semver lo
                         .maxRetryInterval(retryInterval).maxAttempt(CLIENT_RETRY_COUNT)
                         .retryableExceptions(Arrays.asList(DeviceConfigurationException.class)).build();
 
-        try (GreengrassV2DataClient greengrasV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
+        try (GreengrassV2DataClient greengrassV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
                 clientFactory::fetchGreengrassV2DataClient, "get-greengrass-v2-data-client", logger)) {
-            result = greengrasV2DataClient.resolveComponentCandidates(request);
+            result = greengrassV2DataClient.resolveComponentCandidates(request);
         } catch (ResourceNotFoundException e) {
-            logger.atDebug().kv("componentName", componentName).kv("versionRequirements", versionRequirements)
-                    .log("No applicable version found in cloud registry", e);
-            throw new NoAvailableComponentVersionException("No cloud component version satisfies the requirements.",
-                    componentName, versionRequirements);
+            if (e.getMessage().contains("claim platform")) {
```

Comment:
This is not reliable way of checking platform incompatible error. If cloud log message ever changes, this would break.

Instead of log parsing and creating a new error code, I would recommend directly logging the message we get from cloud's `ResourceNotFoundException` into `NoAvailableComponentVersionException`

Reply:
Yes, I agree but I discussed this case with Michael. The message from cloud is only dumped into the logs and the message which is displayed on the console or through cli is what we build in makeMessage(). Having the same version conflict message for incompatible platform claim is not a clear way to explain what really went wrong.

Reply:
The `makeMessage` function also includes the initial message right? so if you throw something like `NoAvailableComponentVersionException("No cloud component version satisfies the requirements due to " + e.getMessage(),
                    componentName, versionRequirements);`, customer can still see it in the console.

Reply:
Yes, but after the initial message, the customer will see `Check whether the version constraints conflict and that the component exists in your AWS account with a version that matches the version constraints.  If the version constraints conflict, revise deployments to resolve the conflict.` That might give away the wrong information about version conflict and not incompatible platform.